### DISCUSSION TITLE
Making HTTP Requests More Agile & Adaptive

### DISCUSSION THREAD

**Discussion by ForeverYoungFly** [logancyang/obsidian-copilot#578] [Ideas]
Hello, thank you for your outstanding work first and foremost. 

However, I've encountered an issue or rather, a suggestion: would it be possible to incorporate an HTTP request model? 

I've built an AI application using the qwen-max model, which embeds my knowledge base. Although custom models were added in version 2.6.0, under this model, I'm unable to invoke a specific AI model application for service via the app_id. Have you considered adding the functionality to invoke models via HTTP POST in the future? This way, users could leverage HTTP to call AI applications they've created within third-party models. 

![Snipaste_2024-09-01_16-46-26](https://github.com/user-attachments/assets/ab649158-1ed7-41ca-a579-906dc3cb148c)

Regardless, your work is highly commendable, and I express my gratitude once again!

**Comment by logancyang** [comment]
@ForeverYoungFly what do you mean by http request? It is using HTTP request. 

- The plugin supports any 3rd party models that has **openai format**. Does your model have openai format?
- Did it work before v2.6.0?
